src/main/java/cs2103/v15_1j/jimjim/command/DeleteCommand.java

```diff
@@ -28,7 +28,12 @@ public char getPrefix() {
     @Override
     public UIFeedback undo(ControllerStates conStates) {
 		// Add task/event back at former position
-	    conStates.masterList.add(taskNum-1, backup);
+	    conStates.masterList.add(backup);
+	    
+	    if(!conStates.searchResultsList.isEmpty()){
```

Comment:
When you undo then, backup would be added back into the masterlist. But if search results were showing, it would return to the masterlist but not appear in search results.